src/Runtime/include/OnnxMlir.h

```diff
@@ -0,0 +1,289 @@
+//===-------- RtMemRef.h - external RtMemRef C/C++ API call header --------===//
+//
+// Copyright 2019-2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains declaration of external RtMemRef data structures and
+// helper functions.
+//
+//===----------------------------------------------------------------------===//
+#ifndef __RTMEMREF_H__
+#define __RTMEMREF_H__
+
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdbool.h>
+#include <stdint.h>
+#endif
+
+/*! \mainpage ONNX-MLIR Model Execution API documentation
+ *
+ * \section intro_sec Introduction
+ *
+ * ONNX-MLIR project comes with an executable `onnx-mlir` capable
+ * of compiling onnx models to a shared library. In this documentation, we
+ * explain and document ways to interact programmatically with the compiled
+ * shared library using the API's defined in `src/Runtime/include/OnnxMlir.h`.
+ *
+ * \section Important Data Structures
+ * \subsection RtMemRef
+ *
+ * `RtMemRef` is the data structure used to describe the runtime information
+ * (rank, shape, data type, etc) associated with a tensor (or `MemRef`, in the
+ * context of MLIR). Specifically, it can be described as a struct with the
+ * following data members:
+ *
+ * ```cpp
+ * void *_data;            // data buffer
+ * void *_alignedData;     // aligned data buffer that the rmr indexes.
+ * INDEX_TYPE _offset;     // offset of 1st element
+ * INDEX_TYPE *_dataSizes; // sizes array
+ * int64_t *_dataStrides;  // strides array
+ * int _dataType;          // ONNX data type
+ * int _rank;              // rank
+ * std::string _name;      // optional name for named access
+ * bool _owningData;       // indicates whether the Rmr owns the memory space
+ *                            referenced by _data. Rmr struct will release the
+ * memory space refereced by _data upon destruction if and only if it owns it.
```

Comment:
Is the user supposed to know about how to use this? No, and I've changed the doc so that we don't reveal things like this for users. Code has always been such that user has no access to internal details. Gong achieved this using forward declarations for API function call deaclarations.